src/library/previewbuttondelegate.cpp

```diff
@@ -26,11 +25,17 @@ PreviewButtonDelegate::PreviewButtonDelegate(QTableView* parent, int column)
     connect(this, SIGNAL(loadTrackToPlayer(TrackPointer, QString, bool)),
             parent, SIGNAL(loadTrackToPlayer(TrackPointer, QString, bool)));
 
-    m_pButton = new QPushButton("", m_pTableView);
+    // The button needs to be parented to receive the parent styles
+    m_pButton = make_parented<QPushButton>(parent);
     m_pButton->setObjectName("LibraryPreviewButton");
     m_pButton->setCheckable(true);
     m_pButton->setChecked(false);
+    // We need to hide the button that it is not painted by the QObject tree    
     m_pButton->hide();
+    // set the visible, to stop resizing in the background 
+    // which may lead to a crash since it is actally visible by the render() call 
```

Comment:
what do you mean by "by"?  is it:

"since it is actually made visible by the render() call"
or
"since it is already visible by the time of the render() call"